File: lua/pn.lua
#!/usr/bin/luajit

local p = true
local n = 100000
print("2\n3")

for i = 5, n, 2 do
    for j = 3, (i / 2) + 1, 2 do
        if i % j == 0 then
            p = false
            break
        else
            p = true
        end
    end

    if p then
        print(i)
        p = false
    end
end

File: ruby/pn.rb
#!/usr/bin/ruby

p = true
n = 100000
puts "2\n3"

for i in (5..n).step(2) do
    for j in (3..(i / 2).to_i) do
        if i % j == 0 then
            p = false
            break
        else
            p = true
        end
    end

    if p then
        puts i
        p = false
    end
end
